single_kernel_mongo/events/lifecycle.py

```diff
@@ -163,6 +168,16 @@ def on_config_changed(self, event: ConfigChangedEvent):
             self.dependent.update_config_and_restart()
         except (UpgradeInProgressError, WaitingForLeaderError):
             event.defer()
+        except InvalidConfigRoleError:
+            logger.info("Invalid config role.")
+            if self.charm.unit.is_leader():
+                self.dependent.state.statuses.add(
+                    MongoDBStatuses.INVALID_ROLE.value,
+                    scope="app",
```

Comment:
nice, I believe this would be our first non-refresh related `app` status in mongodb :)